src/drt/src/dr/FlexDR.cpp

```diff
@@ -1412,9 +1846,11 @@ int FlexDR::main()
       break;
     }
   }
-
   for (auto& args :
        strategy(router_cfg_->ROUTESHAPECOST, router_cfg_->MARKERCOST)) {
+    if (iter_ > router_cfg_->END_ITERATION) {
+      break;
+    }
```

Comment:
Not really. This condition is hit after incrementing iter_ in both cases. It would only differ if END_ITERATION=-1. In that case, master would perform the 0th iteration while this branch would break before.